apps/core-app/src/utils/debounceHook.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from 'react';
+
+function useDebounce<T>(value: T, delay?: number): T {
+  const [debouncedValue, setDebouncedValue] = useState<T>(value);
+
+  useEffect(() => {
+    const timer = setTimeout(() => setDebouncedValue(value), delay || 500);
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [value, delay]);
+
+  return debouncedValue;
+}
```

Comment:
i feel like we shuld move this to the ui library hooks -- seems pretty generic across apps

Reply:
Agreed